Mathlib/Topology/MetricSpace/HausdorffDistance.lean

```diff
@@ -905,6 +905,15 @@ theorem mem_thickening_iff_infEdist_lt : x ∈ thickening δ s ↔ infEdist x s
   Iff.rfl
 #align metric.mem_thickening_iff_inf_edist_lt Metric.mem_thickening_iff_infEdist_lt
 
+/-- An exterior point of a subset `E` (i.e., a point outside the closure of `E`) is not in the
+(open) `δ`-thickening of `E` for small enough positive `δ`. -/
+lemma eventually_not_mem_thickening_of_infEdist_pos {E : Set α} {x : α} (h : x ∉ closure E) :
+    ∀ᶠ δ in 𝓝 (0 : ℝ), x ∉ Metric.thickening δ E := by
+  obtain ⟨ε, ⟨ε_pos, ε_lt⟩⟩ := exists_real_pos_lt_infEdist_of_not_mem_closure h
+  filter_upwards [Iio_mem_nhds ε_pos] with δ hδ
+  simp only [thickening, mem_setOf_eq, not_lt]
+  exact (((ENNReal.ofReal_lt_ofReal_iff ε_pos).mpr hδ).trans ε_lt).le
```

Comment:
Btw, these kinds of terms are a mouthful, and I still often end up writing them. Clearly this is bad for readability --- is there a more stylish way?

(Although, for mathematically obvious lemmas like this and the next, I'm of the opinion that only the readability of the statement is important, while their trivial proofs can be obfuscated/golfed. But I would understand also the desire to write readable terms in the proof... So if there is an easy style improvement, I'll be happy to learn it!)

Reply:
There isn't really any universal trick. I guess this is slightly better:
```lean
  obtain ⟨ε, ⟨ε_pos, ε_lt⟩⟩ := exists_real_pos_lt_infEdist_of_not_mem_closure h
  filter_upwards [eventually_lt_nhds ε_pos] with δ hδ
  simp only [thickening, mem_setOf_eq, not_lt]
  exact le_trans (ENNReal.ofReal_le_ofReal hδ.le) ε_lt.le
```
Note that the reason the term looks better is that the mathematical proof is very slightly simpler (we're just using that `ofReal` is monotone, so no need for `ε_pos` to appear.

About the `filter_upwards` line: technically this does exactly the same as the current code, except that you end up with `hδ : δ < ε` instead of `hδ : δ ∈ Iio ε`, allowing dot notation. In general I would suggest using the `Eventually` version over the `_ ∈ ℱ` version in `filter_upwards` when both exist, but it's not really important.

Reply:
Oh and of course you can also modify the `cthickening` version in a similar way.